h/util/uri.py

```diff
@@ -96,6 +96,11 @@
         #     http://help.webtrends.com/en/analytics10/#qpr_about.html
         #
         r"^WT\..+$",
+        # Amazon security access token. Reference:
+        #
+        #     https://docs.aws.amazon.com/general/latest/gr/sigv4-add-signature-to-request.html
+        #
+        r"^x-amz-security-token$",
```

Comment:
In the linked documentation this query param is `X-Amz-Security-Token`. Note the capitalization. Looking at the JSON POST bodies in the [Sentry reports](https://sentry.io/organizations/hypothesis/issues/1214722226/?referrer=github_integration) associated with [the issue](https://github.com/hypothesis/h/issues/5713) it appears that both upper and lower-case variants of this query param are used by various services. Therefore we need to make sure:

- The comparison here is case insensitive
- We have tests for both variants

`X-Amz-Security-Token` is the canonical capitalization given in the docs.